python/ray/_private/runtime_env/utils.py

```diff
@@ -89,6 +89,10 @@ async def check_output_cmd(
         #     https://docs.python.org/3/library/asyncio-subprocess.html#asyncio.asyncio.subprocess.Process.stderr
         #   * Avoid mixing multiple outputs of concurrent cmds.
         stdout, _ = await proc.communicate()
+    except asyncio.exceptions.CancelledError as e:
+        # since Python 3.9, when cancelled, the inner process needs to throw as it is
+        # for asyncio to timeout properly https://bugs.python.org/issue40607
```

Comment:
What's the issue here? I checked the bug and the fix (https://github.com/python/cpython/pull/20054/files). Seems we can catch CancelledError and raise a different one.

Reply:
The issue is that this code throws as RuntimeError instead of CancelledError. Before https://github.com/python/cpython/commit/382a5635bd10c237c3e23e346b21cde27e48d7fa, regardless of that error this code throw, asyncio/tasks.py will throw TimeoutError which is what we need. After the change, asyncio/tasks.py will throw RuntimeError up in the line


```
except exceptions.CancelledError as exc:
                raise exceptions.TimeoutError() from exc
```

Reply:
I see.